src/webots/gui/WbSimulationView.cpp

```diff
@@ -610,20 +614,20 @@ void WbSimulationView::makeMovie() {
   updateFastModeOverlay();
 }
 
-void WbSimulationView::switchToRunModeIfNecessary() {
+void WbSimulationView::show3DViewIfNecessary() {
   // remove "Fast Mode" overlay if necessary
-  if (WbSimulationState::instance()->isFast()) {
-    WbSimulationState::instance()->setMode(WbSimulationState::RUN);
+  if (!WbSimulationState::instance()->is3DViewShown()) {
+    WbSimulationState::instance()->show3DView(true);
     mView3D->hideFastModeOverlay();
-    mNeedRestoreFastMode = true;
+    mNeedToHide3dView = true;
   }
 }
 
-void WbSimulationView::restoreFastModeIfNecessary() {
-  if (mNeedRestoreFastMode) {
+void WbSimulationView::restore3dViewIfNecessary() {
+  if (mNeedToHide3dView) {
     mView3D->showFastModeOverlay();
-    WbSimulationState::instance()->setMode(WbSimulationState::FAST);
-    mNeedRestoreFastMode = false;
+    WbSimulationState::instance()->show3DView(true);
```

Comment:
Shouldn't this code disable 3D rendering instead?
```suggestion
    WbSimulationState::instance()->show3DView(false);
```